src/port1.0/portbump.tcl

```diff
@@ -0,0 +1,315 @@
+                set calculated_checksums_array($distfile) $calculated_checksums
+
+                if {![regexp {\.html?$} ${distfile}] &&
+                    ![catch {strsed [exec [findBinary file $portutil::autoconf::file_path] $fullpath --brief --mime] {s/;.*$//}} mimetype]
+                    && "text/html" eq $mimetype} {
+                    # file --mime-type would be preferable to file --mime and strsed, but is only available as of Snow Leopard
+                    set wrong_mimetype yes
+                    set htmlfile_path ${fullpath}.html
+                    file rename -force $fullpath $htmlfile_path
+                }
+            }
+        }
+    } else {
+        # Something went wrong with the syntax.
+        return -code error "[msgcat::mc "Unable to verify file checksums"]"
+    }
+
+    if {![tbool mismatch]} {
+        ui_msg "No changes needed."
+        return 0
+    }
+
+    if {[tbool wrong_mimetype]} {
+        # We got an HTML file, though the distfile name does not suggest that one was
+        # expected. Probably a helpful DNS server sent us to its search results page
+        # instead of admitting that the server we asked for doesn't exist, or a mirror that
+        # no longer has the file served its error page with a 200 response.
+        ui_notice "***"
+        ui_notice "The non-matching file appears to be HTML. See this page for possible reasons"
+        ui_notice "for the bump mismatch:"
+        ui_notice "<https://trac.macports.org/wiki/MisbehavingServers>"
+        ui_notice "***"
+        ui_notice "The file has been moved to: $htmlfile_path"
+        
+        return -code error "[msgcat::mc "Unable to verify file checksums"]"
+    } else {
+        # Show the desired checksum line for easy cut-paste
+        # based on the previously calculated values, plus our default types
+        set sums {}
+
+        foreach distfile $all_dist_files {
+            if {[llength $all_dist_files] > 1} {
+                lappend sums $distfile
+            }
+
+            set missing_types $portchecksum::default_checksum_types
+
+            # Append the string for the calculated types and note any of
+            # our default types that were already calculated
+            if {[info exists calculated_checksums_array($distfile)] && [llength $calculated_checksums_array($distfile)]} {
+                set calculated_checksums $calculated_checksums_array($distfile)
+                foreach {type sum} $calculated_checksums {
+                    set found [lsearch -exact ${missing_types} ${type}];
+                    if { ${found} != -1} {
+                        set missing_types [lreplace ${missing_types} ${found} ${found}]
+                    }
+                }
+            }
+
+            # Append the string for any of our default types that were
+            # not previously calculated
+            if {[llength $missing_types]} {
```

Comment:
This block currently does nothing. I'd propose you move it to a separate commit on top and open a further PR for this and we can merge this as-is now.

Reply:
Done, moved the remaining TODOs to separate branches.